x-pack/plugins/ml/server/lib/ml_client/ml_client.ts

```diff
@@ -160,28 +162,44 @@ export function getMlClient(
   return {
     async closeJob(...p: Parameters<MlClient['closeJob']>) {
       await jobIdsCheck('anomaly-detector', p);
-      return mlClient.closeJob(...p);
+      return auditLogger.wrapTask(() => mlClient.closeJob(...p), 'ml_close_ad_job', p);
```

Comment:
Did you consider using [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) instead of manually wrapping every call?

Reply:
I had not, it would probably be just as much code, but could be neater and mean we don't have changes in the ml_client file.
I'm not sure how we'd be able to intercept the calling of the function to log the error if it fails.